src/Sylius/Bundle/AdminBundle/EventListener/AdminFilterSubscriber.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\EventListener;
+
+use Sylius\Bundle\AdminBundle\Storage\FilterStorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class AdminFilterSubscriber implements EventSubscriberInterface
```

Comment:
I wonder, do we need a subscriber for that? The only moment when we need to set the filters is when we press the "Filter" button (which means - we access the `indexAction` with some `criteria` in the query parameters). The moment when we need to remove them is when we access the `indexAction` without criteria in the query parameters. Is it possible to just override/decorate the `indexAction` by default to take it into account? 🤔 We would be able to skip the subscriber then 🚀

Reply:
Second thought :D Probably it would be hard to enhance an `indexAction` for all resources other than changing it in the ResourceBundle itself 💃

Reply:
Final thought 🚀 Let's leave it as it is right now and try to resolve it later her or on ResourceBundle itself 🖖